docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,351 @@
+
+For more details about Azure AD Pod Identity please visit this [link](#https://github.com/Azure/aad-pod-identity)  
+
+## <a name='Motivation'></a>Motivation
+
+AWI gives the capability to federate the credentials with external identity providers in a Kubernetes native way. This approach overcomes several limittations of Azure AD Identity as mentioned below.
+- Removes scale and performance issues that existed for identity assignment.
+- Supports K8s clusters hosted in any cloud or on premise.
+- Supports both Linux and Windows workloads and removes the need for CRD and pods that intercept IMDS traffic.
+
+To learn more about AWI please visit this link https://azure.github.io/azure-workload-identity/docs/introduction.html
+
+### <a name='Goals'></a>Goals
+
+- Integrate Azure Workload Identity into capz.
+
+### <a name='Non-GoalsFutureWork'></a>Non-Goals / Future Work
+
+- Migration to `azidentity` from `adal` module when using `go-autorest` library in capz as `adal` will be deprecated. 
+
+## <a name='Proposal'></a>Proposal
+
+In this model, Kubernetes cluster itself becomes the token issuer issuing tokens to Kubernetes service accounts. These service accounts can be configured to be trusted on Azure AD application or user assigned identity. Workloads/pods can use this service account token which is projected to it's volume and can exchange for an Azure AD token.
+
+The first step for creating K8s cluster using capz is creating a bootstrap cluster which is a Kind cluster and then deploying capi, capz and other helper components. On a high level the workflow looks the following to be able to use AWI and create a management cluster.
+
+**Bootstrap Cluster**
+
+- The operator/admin should be able generate signing key pair or BYO key pair. 
+
+- The bootstrap cluster i.e the Kind K8s cluster should be configured with appropriate flags on kube-apiserver and kube-controller-manager and the key pairs should be mounted on the container path for control plane node. 
+
+- The operator/admin should set up OIDC issuer URL which means having access to azure blob storage where the following two documents will be stored in the blob storage container.
+  - Generate and upload the Discovery Document.
+  - Generate and upload the JWKS Document.
+ 
+- The AWI mutating admission webhook should be installed on the bootstrap cluster.
+
+- A service account should be created in the same namespace as that of capz pod and tagged with the client ID of the Azure AD application or user addigned identity on the bootstrap cluster. The service account should also be associated as a Subject with the appropriate ClusterRoleBindings and RoleBindings for the usual RBAC. This is the service account that the capz pod should use.
+
+- The capz manager service account should be annotated with the Azure AD or User Assigned Client ID. 
+
+- A fedreated credential should be created between the identity and the OIDC issuer URL and the service account(subject). 
```

Comment:
```suggestion
- A federated credential should be created between the identity and the OIDC issuer URL and the service account(subject). 
```

Might be good to add a reference to https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html